array.go

```diff
@@ -913,3 +913,43 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+func (a *Array) Filter(filter func(index int, value *Value) bool) *Array {
```

Comment:
Please add documentation comment with code example to Array.Filter and Object.Filter.

Don't forget to mention important details:

* that function returns new array
* when predicate is considered to be successful and failed
* what happens if assertion inside predicate fails